src/Decompiler/Typing/TypeCollector.cs

```diff
@@ -109,7 +109,11 @@ public void CollectUserGlobalVariableTypes()
             {
                 if (ud.Value.DataType != null)
                 {
-                    var dt = ud.Value.DataType.Accept(deser);
+                    var dt = program.FindGlobalField(ud.Key);
```

Comment:
Can you write a unit tests in `TypeCollectorTests.cs` that illustrates the problem this change is trying to solve?